src/types.ts

```diff
@@ -451,7 +451,11 @@ export interface FunctionOptions {
 export type StepRunResponse =
   | {
       status: 500;
-      error?: string;
+      error?: Record<string, any> | string;
```

Comment:
I don't believe any step is returning a string error anymore with this code change - they should all be objects w/ the new marshal function. Perhaps we could drop the `string` here and in the `400`